### PR TITLE
Remove deprecated name attribute from links and other HTML validation errors

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
Looking at the html validation errors from runner on [this](https://trello.com/c/TBWfGlC5/3652-fix-html-errors-and-warnings) trello card I noticed we still use the name attribute on links in a couple of places. That attribute has been deprecated so this PR is to remove it when links are used in buttons and lists.

- Another HTML validation error I noticed was that we don't have a default for the `<title>` tag. This tag shouldn't be empty so I have added a default of 'ONS Design System' when `page_title` isn't supplied.

- I have also removed the `scrolling` attribute on iframes used for examples. It now deprecated and was only being set to "auto" anyway which is default. 

- And all aria labels would've been able to have been set to `null` have been given defaults

- Also updates docs for these changes, adds `ariaAsc` and `ariaDesc` which were previously undocumented and updates the names of `aria_sort`, `data_sort`, `aria_asc` and `aria_desc` to bring them in line with the naming convention of other variables.

**BREAKING CHANGE** due to the changing of param names

### How to review
- Tests pass
- https://validator.w3.org/ this validator will now not show these errors

### CODE REVIEW

**Comment by rmccar** [comment]
[HTML validation error: name attribute deprecated on links](https://trello.com/c/dVN1e3Fo/1158-html-validation-error-name-attribute-deprecated-on-links)

**Review by boxadesign — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

File: src/components/section-navigation/_macro.njk [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% macro onsSectionNavigation(params) %}
-  <nav class="section-nav" id="{{ params.id }}" aria-label="{{ params.ariaLabel }}" role="menu">
-      <ul class="section-nav__list" aria-label="{{ params.ariaListLabel }}">
+  <nav class="section-nav" id="{{ params.id }}" aria-label="{{ params.ariaLabel | default("Section Navigation") }}" role="menu">
```
Comment (jrbarnes9) [inline src/components/section-navigation/_macro.njk:2]:
```suggestion
  <nav class="section-nav" id="{{ params.id }}" aria-label="{{ params.ariaLabel | default("Section menu") }}" role="menu">
```

File: src/components/section-navigation/_macro.njk [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+      <ul class="section-nav__list" aria-label="{{ params.ariaListLabel | default("Section Links") }}">
```
Comment (jrbarnes9) [inline src/components/section-navigation/_macro.njk:3]:
```suggestion
      <ul class="section-nav__list" aria-label="{{ params.ariaListLabel | default("Section menu links") }}">
```

File: src/views/partials/sub-navigation/_macro.njk [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {% macro patternlibSubNavigation(params) %}
-    <nav class="patternlib-subnav" aria-label="{{ params.ariaLabel }}">
-        <ul class="patternlib-subnav__list list list--bare" aria-label="{{ params.ariaListLabel }}">
+    <nav class="patternlib-subnav" aria-label="{{ params.ariaLabel | default("Sub Navigation") }}">
```
Comment (jrbarnes9) [inline src/views/partials/sub-navigation/_macro.njk:2]:
```suggestion
    <nav class="patternlib-subnav" aria-label="{{ params.ariaLabel | default("Sections menu") }}">
```

File: src/views/partials/sub-navigation/_macro.njk [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
+        <ul class="patternlib-subnav__list list list--bare" aria-label="{{ params.ariaListLabel | default("Sub Links") }}">
```
Comment (jrbarnes9) [inline src/views/partials/sub-navigation/_macro.njk:3]:
```suggestion
        <ul class="patternlib-subnav__list list list--bare" aria-label="{{ params.ariaListLabel | default("Sections menu links") }}">
```

**Review by jrbarnes9 — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Pull request merged**